codec-http/src/test/java/io/netty/handler/codec/http/cookie/ServerCookieEncoderTest.java

```diff
@@ -60,4 +63,29 @@ public void testEncodingWithNoCookies() {
         assertNotNull(encodedCookie2);
         assertTrue(encodedCookie2.isEmpty());
     }
+
+    @Test
+    public void testEncodingMultipleCookiesStrict() {
+        List<String> result = new ArrayList<String>();
+        result.add("cookie2=value2");
+        result.add("cookie1=value3");
+        Cookie cookie1 = new DefaultCookie("cookie1", "value1");
+        Cookie cookie2 = new DefaultCookie("cookie2", "value2");
+        Cookie cookie3 = new DefaultCookie("cookie1", "value3");
+        List<String> encodedCookies = ServerCookieEncoder.STRICT.encode(cookie1, cookie2, cookie3);
+        assertEquals(result, encodedCookies);
+    }
+
+    @Test
+    public void testEncodingMultipleCookiesLax() {
+        List<String> result = new ArrayList<String>();
+        result.add("cookie1=value1");
+        result.add("cookie2=value2");
+        result.add("cookie1=value3");
+        Cookie cookie1 = new DefaultCookie("cookie1", "value1");
+        Cookie cookie2 = new DefaultCookie("cookie2", "value2");
+        Cookie cookie3 = new DefaultCookie("cookie1", "value3");
+        List<String> encodedCookies = ServerCookieEncoder.LAX.encode(cookie1, cookie2, cookie3);
+        assertEquals(result, encodedCookies);
+    }
```

Comment:
This is for server cookie encoding, not client encoding. There are two different things going on in this PR:

1. For client encoding, paths are sorted in decreasing order of path length in strict mode.
2. For server encoding, cookies are not sorted in strict mode, but if there is more than one cookie with the same name, all but the last instance is dropped (i.e. the last-set cookie with a matching name is the only one that is encoded in the header). This is another judgment call I made, since the spec does not say which cookie should be chosen in the case where there are multiple cookies with the same name, it only says that only one of them should be sent. There may be a better heuristic involving paths or path lengths, but picking the last-set cookie of a given name is arguably as reasonable as anything else.

This test does test case 2. The first test I added is for strict mode, the second test is for lax mode.